src/preprocessing/split.py

```diff
@@ -48,4 +48,32 @@ def split_into_exclusive_datasets(datasources_dir="datasources/processed_data",
         subset = torch.utils.data.random_split(full_subset, subset_split)
         subsets.append(subset)
 
-    return subsets
\ No newline at end of file
+    return subsets
+
+def split_test_data_into_3_seconds(datasources_dir="datasources"):
+    print(f'Splitting test music Data into 3 second chunks')
+    error_converted_files = []
+    data_dir = os.path.join(datasources_dir, "test_data_wav")
+    processed_dir = os.path.join(datasources_dir,"processed_test_data")
+    if not os.path.isdir(processed_dir):
+        os.mkdir(processed_dir)
+
+    for _, dirs, _ in os.walk(data_dir):
+        for dir in dirs:
+            print(f"Traversing {dir}")
+            origin_dir = os.path.join(data_dir, dir)
+            destination_dir = os.path.join(processed_dir, dir)
+            if not os.path.isdir(destination_dir):
+                os.mkdir(destination_dir)
+            for _, _, files in os.walk(origin_dir):
+                for file in files:
+                    origin_file = os.path.join(origin_dir, file)
+                    try:
+                        sound = AudioSegment.from_wav(origin_file)
+                    except:
+                        print(f'Skipping {origin_file}')
+                        continue
+                    num_segments = int(45 / 3) #Skip first 10 seconds and save 45 seconds worth
```

Comment:
Nice idea!

Reply:
out of curiosity, why are we skipping the first 10 seconds?

Reply:
Also, are all the data 3 seconds long? Could there be scenarios where there are different lengths of data? @tmahabir

Reply:
@jackhuang-intel 
We are skipping the first 10 seconds because I found some songs would take a while to start actually playing music. Thus, if we didn't skip the first 10 seconds, there would be samples with audio that would skew what that particular genre sounds like.

As per your second question, I ensured that all songs I downloaded were a minimum length of 1min. Thus, if we skip 10 seconds of it and then save the next 45 seconds of the song, we can be confident that we will get 15 samples of 3 seconds length for each song.

Reply:
ah I see. And from these 45 seconds we get can get fifteen 3-second samples, right?